test/ex_doc/language/elixir_test.exs

```diff
@@ -500,7 +500,9 @@ defmodule ExDoc.Language.ElixirTest do
     ExDoc.Refs.insert([
       {{:module, AutolinkTest.Foo}, :public},
       {{:function, AutolinkTest.Foo, :bar, 1}, :hidden},
-      {{:type, AutolinkTest.Foo, :bad, 0}, :hidden}
+      {{:type, AutolinkTest.Foo, :bad, 0}, :hidden},
+      {{:module, AutoLinkTest.Hidden}, :hidden},
+      {{:type, AutoLinkTest.Hidden, :my_type, 0}, :hidden}
```

Comment:
note to self: The testing setup is not fantastic. I don't know off the top of my head whether _here_ :hidden is supposed to reflect the type is hidden or the type is (so to speak) public but because it is in hidden module it ends up hidden. This is an implementation detail.

A more robust way to test this would be more end-to-end:

<img width="848" alt="image" src="https://github.com/user-attachments/assets/081a3994-1232-457b-95e2-3aa06d37a340">